### PR TITLE
New Project Screen

### PR DESCRIPTION
**Pull request by masterrr**
Closes https://github.com/toggl/mobile/issues/574
Closes https://github.com/toggl/mobile/issues/573
Closes https://github.com/toggl/mobile/issues/572
screenshot_2015-04-06-21-59-42

### CODE REVIEW

File: Joey/UI/Views/TogglField.cs [hunk 1]
```diff
@@ -5,16 +5,15 @@
 
 namespace Toggl.Joey.UI.Views
 {
-    public class EditTimeEntryBit : RelativeLayout
+    public class TogglField : RelativeLayout
```
Comment (monday8am) [inline Joey/UI/Views/TogglField.cs]: :+1: this name is better IMHO

File: Joey/Resources/layout/NewProjectFragment.axml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        android:id="@+id/NewProjectsFragmentToolbar"
+        android:layout_width="match_parent"
+        android:layout_height="?android:attr/actionBarSize"
+        android:background="@null"
+        android:theme="@style/ThemeOverlay.AppCompat.ActionBar"
+        android:popupTheme="@style/ThemeOverlay.AppCompat.Dark">
+        <Button 
+        android:id="@+id/NewProjectsFragmentSaveButton"
+        android:text="@string/NewProjectSaveButtonText"
+        android:background="?android:attr/selectableItemBackgroundBorderless"
```
Comment (monday8am) [inline Joey/Resources/layout/NewProjectFragment.axml]: @masterrr  this value "android:attr/..." is supported only in 21+ API levels. You should create an style for this button and add it to values/styles and values-21/styles. The second folder will be used by Lollipop devices, then you can use this kind of values.

File: Joey/UI/Activities/NewProjectActivity.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+using Toggl.Joey.UI.Fragments;
+using Toggl.Phoebe.Data.Models;
+
+using Activity = Android.Support.V7.App.ActionBarActivity;
+using Fragment = Android.Support.V4.App.Fragment;
+
+namespace Toggl.Joey.UI.Activities
+{
+    [Activity (Label = "NewProjectActivity")]
+    public class NewProjectActivity : Activity
```
Comment (monday8am) [inline Joey/UI/Activities/NewProjectActivity.cs]: All activities must extends BaseActivity
https://github.com/toggl/mobile/blob/master/Joey/UI/Activities/BaseActivity.cs

File: Joey/UI/Activities/NewProjectActivity.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        public static readonly string ExtraWorkspaceId = "com.toggl.timer.workspace_id";
+
+        private WorkspaceModel workspace;
+
+        protected override void OnCreate (Bundle bundle)
+        {
+            SetTheme (Resource.Style.Theme_AppCompat_Light_NoActionBar);
```
Comment (monday8am) [inline Joey/UI/Activities/NewProjectActivity.cs]: Create an specific theme for the activity or use an existing theme:
example: https://github.com/toggl/mobile/blob/master/Joey/Resources/values/Styles.xml#L11

File: Joey/UI/Fragments/NewProjectFragment.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        public override void OnStart ()
+        {
+            base.OnStart ();
+            var inputService = (InputMethodManager)Activity.GetSystemService (Context.InputMethodService);
+            ProjectBit.TextField.PostDelayed (delegate {
+                inputService.ShowSoftInput (ProjectBit.TextField, ShowFlags.Implicit);
+            }, 100);
+        }
+
+        async void SaveButtonHandler (object sender, EventArgs e)
```
Comment (monday8am) [inline Joey/UI/Fragments/NewProjectFragment.cs]: ```private``` scope should be added in all private methods. I know is not needed but is part of our code style.

**Comment by monday8am** [comment]
SetHint here?
https://github.com/toggl/mobile/blob/dmitry-new-proj-screen/Joey/UI/Views/TogglField.cs#L54

**Comment by monday8am** [comment]
@masterrr is a good advance. Please, remove the client field and all related stuff (will be added the next release)  and fix the previous comments.

File: Joey/UI/Activities/NewProjectActivity.cs (hunk 3 above)
Comment (masterrr) [inline Joey/UI/Activities/NewProjectActivity.cs]: Won't do until https://github.com/toggl/mobile/pull/587 is merged in. I really need ActionBarActivity there :smile:

File: Joey/UI/Activities/NewProjectActivity.cs (hunk 4 above)
Comment (masterrr) [inline Joey/UI/Activities/NewProjectActivity.cs]: Won't do until #587 is merged in. It'll be there https://github.com/toggl/mobile/pull/587/files#diff-1ff5a6fe48d5977e865d7540f90420c7R3.

File: Joey/Resources/layout/NewProjectFragment.axml (hunk 2 above)
Comment (masterrr) [inline Joey/Resources/layout/NewProjectFragment.axml]: @monday8am we do not have values-21 folder now, should I create values-21 folder and copy all files (https://github.com/toggl/mobile/tree/master/Joey/Resources/values) over? Or can I just create styles.xml in values-21, and styles I define override (if name/namespace collision) those in “values”?

**Comment by masterrr** [comment]
> SetHint here?
https://github.com/toggl/mobile/blob/dmitry-new-proj-screen/Joey/UI/Views/TogglField.cs#L54
Right. Fixed

File: Joey/Resources/layout/NewProjectFragment.axml (hunk 2 above)
Comment (monday8am) [inline Joey/Resources/layout/NewProjectFragment.axml]: this folder is already added, but in a PR that is not merged (@taavirehemagi :P )
https://github.com/toggl/mobile/pull/587/files#diff-e84dc386d3021693356cf95d12878c3cR650

**Comment by masterrr** [comment]
@monday8am I kept that save button background ripple solution simple, just did it as a native, appcompat manages the background the right way now :) https://github.com/toggl/mobile/commit/9f6ccfa50ce63c03ba431fe1c21d2b9afe28835a

Though the save button can be a little off and text size is small. We need to adjust it in themes after #587 gets merged.

File: Joey/Joey.csproj [hunk 6]
```diff
@@ -622,6 +624,12 @@
     <AndroidResource Include="Resources\drawable-xxhdpi\IcNavReports.png" />
     <AndroidResource Include="Resources\drawable-xxhdpi\IcNavSettings.png" />
     <AndroidResource Include="Resources\drawable-xxhdpi\IcNavTimer.png" />
+    <AndroidResource Include="Resources\menu\ProjectFragmentClientItemMenu.xml" />
```
Comment (taavirehemagi) [inline Joey/Joey.csproj]: I think this file doesn't really exist. Gives an error while building for me.

Comment (monday8am) [inline Joey/Joey.csproj]: @masterrr For me the same.

File: Joey/UI/Activities/NewProjectActivity.cs (hunk 4 above)
Comment (monday8am) [inline Joey/UI/Activities/NewProjectActivity.cs]: @masterrr  #587 is merged.

**Comment by masterrr** [comment]
All fixed! :smile_cat:

**Comment by taavirehemagi** [comment]
Yep, is good now! :+1:

**Comment by monday8am** [comment]
:+1:

**Pull request merged**